src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -925,19 +925,24 @@ bool emberAfTimeSynchronizationClusterSetTimeZoneCallback(
         }
         else
         {
-            TimeState dstState = TimeSynchronizationServer::Instance().UpdateDSTOffsetState();
-            TimeSynchronizationServer::Instance().ClearDSTOffset();
-            if (dstState == TimeState::kActive || dstState == TimeState::kChanged)
-            {
-                emitDSTStatusEvent(commandPath.mEndpointId, false);
-            }
             response.DSTOffsetRequired = true;
         }
     }
     else
     {
         response.DSTOffsetRequired = true;
```

Comment:
This codepath needs to be exercised in a test....

Reply:
If you mean the DSTOffsetRequired, it's tested in cert tests. If you mean the status event cert test timesync 2.11 should hit this path IIUC the states - it checks for dst status events as the dst goes on and off. To be fair, it's failing currently, but the test exists.

Reply:
@cecille @fessehaeve I mean that we have a code change here, which is fixing an issue, but there are no test changes.  That means this code is not being exercised in (automated) tests.

It's possible there are manual cert tests for this, but ideally we would have automated tests here...

Reply:
The gap is not the test, it's the example apps. Static zap means we're not covering all cases in CI. Moving to dynamic app testing would be lovely but is a SDK wide problem